Extensions/xkit_patches.js

```diff
@@ -1,32 +1,25 @@
 //* TITLE XKit Patches **//
-//* VERSION 7.1.2 **//
+//* VERSION 7.1.3 **//
 //* DESCRIPTION Patches framework **//
 //* DEVELOPER new-xkit **//
 
 XKit.extensions.xkit_patches = new Object({
 
 	running: false,
 
-	preferences: {
-		debug_mode: {
-			text: "XKit Developer Mode",
-			default: true,
-			value: true
-		}
-	},
-
 	run: function() {
 		this.running = true;
 
 		var to_run = [];
-		for (var i in this.patches) {
-			to_run.unshift(i);
-			if (i === XKit.version) {
+		for (let i of this.run_order.reverse()) {
```

Comment:
ideally this would be something like `relevant_versions = _.filter(run_order, i => semver_lt(XKit.version, i))` but that's maybe hard to implement. Means we don't have to worry about our control flow being based on loops and breaks though.

Reply:
it's possible but i'm not sure if it's easy to see what it's doing or not since it has to do a semver comparison the long way round